### ISSUE TITLE
Warning after Home Assistant upgrade to 2021.12.0

### ISSUE THREAD

**Issue by pabsi** [rospogrigio/localtuya#658]
## The problem
After upgrading Home Assistant to the latest version (2021.12.0) I now get this in the logs when starting up the server:

```
WARNING (MainThread) [homeassistant.helpers.entity] Entity switch.tuyaswitch (<class 'custom_components.localtuya.switch.LocaltuyaSwitch'>) implements device_state_attributes. Please report it to the custom component author.
```

## Environment
- Localtuya version: Latest release as of today (tag v.3.2.4)
- Last working localtuya version (if known and relevant): n/a/
- Home Assistant Core version: 2021.12.0 
- [] Are you using the Home Assistant Tuya Cloud component ?
- [] Are you using the Tuya App in parallel ?

**Comment by pabsi** [comment]
Definitely not a bug, but can't remove the label :( 
sorry

**Comment by Mangomc** [comment]
Sorry but its defintely the old wrong way you use, so its a bug, Please read the changelog of the new build. And look at https://www.home-assistant.io/docs/configuration/templating/ the yellow warning. Also ur readme is wrong in the way u tell to use the device attributes. 

Avoid using states.sensor.temperature.state, instead use states('sensor.temperature')

It now should be that way:

platform: template
  sensors:
    washing_machine_current:
        friendly_name_template: "Waschmaschine Stromstärke"
        value_template: "{{ state_attr('switch.steckdose_waschmaschine', 'current') | float(0.0) }}"
        unit_of_measurement: 'mA'
        
    washing_machine_current_consumption:
        friendly_name_template: "Waschmaschine aktueller Verbrauch"
        value_template: "{{ state_attr('switch.steckdose_waschmaschine', 'current_consumption') | float(0.0) }}"
        unit_of_measurement: 'W'
        
    washing_machine_voltage:
        friendly_name_template: "Waschmaschine Volt"
        value_template: "{{ state_attr('switch.steckdose_waschmaschine', 'voltage') | float(0.0) }}"
        unit_of_measurement: 'V'


Please fix your integration, i love it :)

**Comment by pabsi** [comment]
> Sorry but its defintely the old wrong way you use, so its a bug, Please read the changelog of the new build. And look at https://www.home-assistant.io/docs/configuration/templating/ the yellow warning. Also ur readme is wrong in the way u tell to use the device attributes.
> 
> Avoid using states.sensor.temperature.state, instead use states('sensor.temperature')
> 
> It now should be that way:
> 
> ```
> platform: template
>   sensors:
>     washing_machine_current:
>         friendly_name_template: "Waschmaschine Stromstärke"
>         value_template: "{{ state_attr('switch.steckdose_waschmaschine', 'current') | float(0.0) }}"
>         unit_of_measurement: 'mA'
>         
>     washing_machine_current_consumption:
>         friendly_name_template: "Waschmaschine aktueller Verbrauch"
>         value_template: "{{ state_attr('switch.steckdose_waschmaschine', 'current_consumption') | float(0.0) }}"
>         unit_of_measurement: 'W'
>         
>     washing_machine_voltage:
>         friendly_name_template: "Waschmaschine Volt"
>         value_template: "{{ state_attr('switch.steckdose_waschmaschine', 'voltage') | float(0.0) }}"
>         unit_of_measurement: 'V'
> ```
> 
> Please fix your integration, i love it :)

Hey, I am not sure whether you are talking to me, or the author of the intergration :sweat_smile: 

I definitely used the custom component the right way as stated in the README (via config flow). It all worked good and without issues until HA updated to 2021.12.

Hope the author fixes this (I'd be happy to contribute with a PR, but I am not familiar enough with neither custom components nor python stuff in HA)

**Comment by Mangomc** [comment]
To the author of the integration but i dont want to create a new issue with the same context

**Comment by jsBlackBelt** [comment]
I am seeing the same issue on HA 2021.12.1 with latest local tuya 3.2.4

**Comment by danielbrunt57** [comment]
Here is the full log message I see:
```
Logger: homeassistant.helpers.entity
Source: helpers/entity.py:549
First occurred: 3:45:55 PM (7 occurrences)
Last logged: 3:49:38 PM

Entity switch.plug_1 (<class 'custom_components.localtuya.switch.LocaltuyaSwitch'>) implements device_state_attributes. Please report it to the custom component author.
Entity switch.plug_3 (<class 'custom_components.localtuya.switch.LocaltuyaSwitch'>) implements device_state_attributes. Please report it to the custom component author.
```
FEI, the browser_mod custom component has the same new issue:
```
Entity sensor.browser_mod_e11f3fb6_438343f8 (<class 'custom_components.browser_mod.sensor.BrowserModSensor'>) implements device_state_attributes. Please report it to the custom component author.
Entity media_player.browser_mod_e11f3fb6_438343f8 (<class 'custom_components.browser_mod.media_player.BrowserModPlayer'>) implements device_state_attributes. Please report it to the custom component author.
Entity light.browser_mod_e11f3fb6_438343f8 (<class 'custom_components.browser_mod.light.BrowserModLight'>) implements device_state_attributes. Please report it to the custom component author.
```

**Comment by danielbrunt57** [comment]
See https://github.com/jsBlackBelt/localtuya/commit/bfba2d7f722b2bacc221f9da3d4beba4b7e1417a

**Comment by jouster1974** [comment]
above fix didnt seem to work...

**Comment by danielbrunt57** [comment]
Don't know what to say.  It worked fine for me...the warning is gone.

![image](https://user-images.githubusercontent.com/49846893/146476190-90cea4e2-fb4c-4db4-8569-391b84a7c40d.png)

**Comment by cihantok** [comment]
> Düzeltme yolda:. [jsBlackBelt@ bfba2d7](https://github.com/jsBlackBelt/localtuya/commit/bfba2d7f722b2bacc221f9da3d4beba4b7e1417a)
> 
> ![resim](https://user-images.githubusercontent.com/49846893/146469238-cf9ba4f3-b4ea-4344-b774-e5b0eb2d6d48.png)
> 
> Düzeltmeyi kendiniz yapabilirsiniz. config/custom_components/localtuya/switch.py ​​satır # 51'i düzenleyin

This worked for me thank you

**Comment by blavak68** [comment]
> > Düzeltme yolda:. [jsBlackBelt@ bfba2d7](https://github.com/jsBlackBelt/localtuya/commit/bfba2d7f722b2bacc221f9da3d4beba4b7e1417a)
> > ![resim](https://user-images.githubusercontent.com/49846893/146469238-cf9ba4f3-b4ea-4344-b774-e5b0eb2d6d48.png)
> > Düzeltmeyi kendiniz yapabilirsiniz. config/custom_components/localtuya/switch.py ​​satır # 51'i düzenleyin
> 
> This worked for me thank you

**yes, this worked for me too, thx**

**Comment by jeremysherriff** [comment]
Fixed in #663

**Comment by pabsi** [comment]
Sorry about this still open issue....
closing now
